packages/shared/src/components/CustomFeedOptionsMenu.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { type ReactElement } from 'react';
+import classNames from 'classnames';
+import { ContextMenuIds } from '../hooks/constants';
+import type { MenuItemProps } from './fields/ContextMenu';
+import ContextMenu from './fields/ContextMenu';
+import { HashtagIcon, MenuIcon as DotsIcon, ShareIcon } from './icons';
+import { MenuIcon } from './MenuIcon';
+import useContextMenu from '../hooks/useContextMenu';
+import { Button, ButtonSize, ButtonVariant } from './buttons/Button';
+import {
+  useShareOrCopyLink,
+  type UseShareOrCopyLinkProps,
+} from '../hooks/useShareOrCopyLink';
+
+type CustomFeedOptionsMenuProps = {
+  className?: string;
+  shareProps: UseShareOrCopyLinkProps;
+};
+
+const CustomFeedOptionsMenu = ({
+  className,
+  shareProps,
+}: CustomFeedOptionsMenuProps): ReactElement => {
+  const [, onShareOrCopyLink] = useShareOrCopyLink(shareProps);
+  const { isOpen, onMenuClick } = useContextMenu({
+    id: ContextMenuIds.CustomFeedContext,
+  });
+
+  const options: MenuItemProps[] = [
+    {
+      icon: <MenuIcon Icon={ShareIcon} />,
+      label: 'Share',
+      action: () => onShareOrCopyLink(),
```

Comment:
This is not doing anything for me on tag page?

Reply:
@rebelchris So apparently it was because webappurl is just "/" and we don't handle that case, so it didn't have a valid url as tag doesn't have a permalink

I swapped it out with location.href, although in preview it is still getting an error for "invalid url", presumably because the preview origin is not an approved domain. Not erroring on local dev.

If there's preferred way to always get the full url, let me know and I'll swap it.

Reply:
Hmm yeah webappUrl is fully qualified in production I think.
So should be fine to leave it then?